docs/letsencrypt.md

```diff
@@ -0,0 +1,125 @@
+below.
+
+DNS-01 verification is also supported, and guidance is offered below.
+
+Let's Encrypt HTTP-01 Requirements
+----------------------------------
+
+Let's Encrypt HTTP-01 verification requires that:
+
+1. The server be accessible over the internet on both ports 80, and 443
+  (and 443 per pulp_installer's design.) The Pulp 3 Ansible Installer will configure the OS's firewall, but any
+  firewall/router in front of it (such as a security group on a cloud provider) must allow them.
+
+2. The server has a DNS name it can be reached at, such as pulp.example.com . This is what must be
+   filled in at `<< domain name >>` in the example playbook below.  Note that many
+   auto-generated DNS names by cloud providers are blocked per Let's Encrypt policy.
+
+These requirements must be satisfied before running The Pulp 3 Ansible Installer.
+
+Let's Encrypt HTTP-01 Example Playbook
+--------------------------------------
+As an example, we are going to write a playbook for installing `pulp_file`, with Let's Encrypt and
+HTTP 01 verification.
+
+The 3rd-party role listed below,
+[lexa-uw.letsencrypt](https://galaxy.ansible.com/lexa-uw/letsencrypt), is an example. The variables listed below are meant for it, and can serve as a guide for other roles.
+
+You can learn more about the variables on the [roles section](https://pulp-installer.readthedocs.io/en/latest/roles/pulp/#role-variables)
+
+1 -  Install the `pulp_installer` collection:
+```
+ansible-galaxy collection install pulp.pulp_installer
+```
+
+2 -  Install the `geerlingguy.postgresql` role:
+```
+ansible-galaxy install geerlingguy.postgresql
+```
+
+3 - Install your preferred 3rd-party role from [Ansible
+Galaxy](https://galaxy.ansible.com/search?deprecated=false&keywords=acme&order_by=-relevance&page=1).
+
+For the example of the role in the playbook below:
+```
+ansible-galaxy install lexa-uw.letsencrypt
+```
+
+4 - Write the following playbook:
+```
+vim install.yml
+```
+
+```yaml
+---
+- hosts: << domain name >>
+  vars:
+    pulp_webserver_httpd_servername: "{{ inventory_hostname }}"
+    lets_encrypt_hostname: "{{ inventory_hostname }}"
+    lets_encrypt_directories_certs: "/etc/letsencrypt"
+    lets_encrypt_directories_data: "/var/lib/pulp/pulpcore_static"
+    pulp_default_admin_password: << YOUR PASSWORD HERE >>
+    pulp_install_plugins:
+      # galaxy-ng: {}
+      # pulp-ansible: {}
+      # pulp-certguard: {}
+      # pulp-container: {}
+      # pulp-cookbook: {}
+      # pulp-deb: {}
+      pulp-file: {}
+      # pulp-gem: {}
+      # pulp-maven: {}
+      # pulp-npm: {}
+      # pulp-python: {}
+      # pulp-rpm: {}
+    pulp_settings:
+      secret_key: << YOUR SECRET HERE >>
+      content_origin: "https://{{ inventory_hostname }}"
+  roles:
+    # Includes running pulp_webserver. letsencrypt depends on a webserver
+    # that can host the .well-known directory.
+    - pulp.pulp_installer.pulp_all_services
+    - role: lexa-uw.letsencrypt
```

Comment:
How come you use different formats for specifying these two roles?

Reply:
Because the 2nd time, we have to apply `become` to it. You can only apply become via the latter format.

I use the usual format for `pulp_all_services because` it's what we specify in our usual playbook, like in quickstart.md.

If you think it makes this playbook easier to read, I'll switch to `role: pulp_all_services`.

AFAIK, there is no difference in behavior between `rolename` and `role: rolename`.

Reply:
What you say makes sense. No need to change it.